testsuite/integration/elytron/src/test/java/org/wildfly/test/integration/elytron/principaltransformers/CasePrincipalTransformerTestCase.java

```diff
@@ -78,104 +99,154 @@ public static WebArchive createDeployment() {
-            client = managementClient;
             try (CLIWrapper cli = new CLIWrapper(true)) {
-                domain = PropertyFileBasedDomain.builder().withName(ELYTRON_SECURITY)
-                        .withUser(USER, PASSWORD, ROLE)
-                        .build();
-                domain.create(cli);
-                cli.sendLine(String.format(
-                        "/subsystem=elytron/case-principal-transformer=%s:add(upper-case=%s)",
-                        PRINCIPAL_TRANSFORMER, true));
-                cli.sendLine(String.format(
-                        "/subsystem=elytron/security-domain=%s:write-attribute(name=realms[0].principal-transformer,value=%s)",
-                        ELYTRON_SECURITY, PRINCIPAL_TRANSFORMER));
-                cli.sendLine(String.format(
-                        "/subsystem=elytron/http-authentication-factory=%1$s:add(http-server-mechanism-factory=%2$s,security-domain=%1$s,"
-                                + "mechanism-configurations=[{mechanism-name=BASIC,mechanism-realm-configurations=[{realm-name=\"%1$s\"}]}])",
-                        ELYTRON_SECURITY, PREDEFINED_HTTP_SERVER_MECHANISM_FACTORY));
-                cli.sendLine(String.format(
-                        "/subsystem=undertow/application-security-domain=%s:add(http-authentication-factory=%s)",
-                        DEP_SECURITY_DOMAIN, ELYTRON_SECURITY));
+                // Create all 3 testing domains
+                domainWithUpperTransformer = createDomain(cli, UPPER_TRANSFORMER_DOMAIN_NAME, UPPER_USER, PASSWORD, ROLE);
+                domainWithLowerTransformer = createDomain(cli, LOWER_TRANSFORMER_DOMAIN_NAME, LOWER_USER, PASSWORD, ROLE);
+                domainWithoutTransformer = createDomain(cli, WITHOUT_TRANSFORMER_DOMAIN_NAME, UPPER_USER, PASSWORD, ROLE);
+
+                // Create upper and lower case transformers
+                createCasePrincipalTransformer(cli, UPPER_PRINCIPAL_TRANSFORMER, true);
+                createCasePrincipalTransformer(cli, LOWER_PRINCIPAL_TRANSFORMER, false);
+
+                // Add transformers to corresponding domains
+                addTransformerToDomain(cli, UPPER_TRANSFORMER_DOMAIN_NAME, UPPER_PRINCIPAL_TRANSFORMER);
+                addTransformerToDomain(cli, LOWER_TRANSFORMER_DOMAIN_NAME, LOWER_PRINCIPAL_TRANSFORMER);
+
+                addHTTPAuthenticationFactory(cli, UPPER_TRANSFORMER_DOMAIN_NAME);
+                addHTTPAuthenticationFactory(cli, LOWER_TRANSFORMER_DOMAIN_NAME);
+                addHTTPAuthenticationFactory(cli, WITHOUT_TRANSFORMER_DOMAIN_NAME);
+
+                addSecurityDomainToUndertow(cli, UPPER_CASE_SECURITY_DOMAIN, UPPER_TRANSFORMER_DOMAIN_NAME);
+                addSecurityDomainToUndertow(cli, WITHOUT_TRANSFORMER_SECURITY_DOMAIN, WITHOUT_TRANSFORMER_DOMAIN_NAME);
+                addSecurityDomainToUndertow(cli, LOWER_CASE_SECURITY_DOMAIN, LOWER_TRANSFORMER_DOMAIN_NAME);
+
```

Comment:
To simplify things slightly, what do you think about having a helper method that creates a domain, optionally adds a transformer to that domain, and adds the application security domain mapping to the Undertow config? That way, instead of calling each of these helper methods various times, we could just call that single method for each domain?

Reply:
Thanks for the feedback! I refactored everything into one method in the latest commit